src/test/java/apoc/export/json/ExportJsonTest.java

```diff
@@ -0,0 +1,252 @@
+        File output = new File(directory, filename);
+        String query = "MATCH (n) return count(n)";
+        TestUtil.testCall(db, "CALL apoc.export.json.query({query},{file})", map("file", output.getAbsolutePath(),"query",query),
+                (r) -> {
+                    assertEquals(true,r.get("source").toString().contains("kernelTransaction: cols(1)"));
+                    assertEquals(output.getAbsolutePath(), r.get("file"));
+                    assertEquals("json", r.get("format"));
+                });
+        assertEquals(jsonResults.get("query_nodes_count"), new Scanner(output).useDelimiter("\\Z").next());
+    }
+
+    @Test
+    public void testExportData() throws Exception {
+        String filename = "data.json";
+        File output = new File(directory, filename);
+        TestUtil.testCall(db, "MATCH (nod:User) " +
+                        "MATCH ()-[reels:KNOWS]->() " +
+                        "WITH collect(nod) as a, collect(reels) as b "+
+                        "CALL apoc.export.json.data(a, b, {file}, null) " +
+                        "YIELD nodes, relationships, properties, file, source,format, time " +
+                        "RETURN *", map("file", output.getAbsolutePath()),
+                (r) -> {
+                    assertEquals(output.getAbsolutePath(), r.get("file"));
+                    assertEquals("json", r.get("format"));
+                });
+        assertEquals(jsonResults.get("data"), new Scanner(output).useDelimiter("\\Z").next());
+    }
+
+    @Test
+    public void testExportDataPath() throws Exception {
+        String filename = "query_nodes_path.json";
+        File output = new File(directory, filename);
+        String query = "MATCH p = (u:User)-[rel]->(u2:User) return u, rel, u2, p, u.id, u.name";
+        TestUtil.testCall(db, "CALL apoc.export.json.query({query},{file})", map("file", output.getAbsolutePath(),"query",query),
+                (r) -> {
+                    assertEquals(output.getAbsolutePath(), r.get("file"));
+                    assertEquals("json", r.get("format"));
+                });
+        assertEquals(jsonResults.get("query_nodes_path"), new Scanner(output).useDelimiter("\\Z").next());
+    }
+
+    private void assertResults(File output, Map<String, Object> r, final String source) {
+        assertEquals(3L, r.get("nodes"));
+        assertEquals(1L, r.get("relationships"));
+        assertEquals(7L, r.get("properties"));
+        assertEquals(source + ": nodes(3), rels(1)", r.get("source"));
+        assertEquals(output.getAbsolutePath(), r.get("file"));
+        assertEquals("json", r.get("format"));
+        assertEquals(true, ((long) r.get("time")) >= 0);
+    }
+
+    private static void populateJsonResultsMap(){
+
+        jsonResults = new HashMap<>();
+
+        jsonResults.put("all", "{\"type\":\"node\",\"id\":0,\"labels\":[\"User\"],\"data\":{\"name\":\"foo\",\"age\":42,\"male\":true,\"kids\":[\"a\",\"b\",\"c\"]}}\n" +
```

Comment:
I actually meant to populate them from regular maps and lists to avoid all these escaped quotes